config/archive_test.go

```diff
@@ -0,0 +1,120 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/contiv/volplugin/watch"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testPolicyName = "foobar"
+)
+
+func revisionCount(s *configSuite) int {
+	revisions, _ := s.tlc.ListPolicyRevisions(testPolicyName)
+	return len(revisions)
+}
+
+func createTestPolicy(s *configSuite, c *C) {
+	err := s.tlc.CreatePolicyRevision(testPolicyName, "{}")
+	c.Assert(err, IsNil)
+}
+
+func (s *configSuite) TestCreatePolicyRevision(c *C) {
+	c.Assert(revisionCount(s), Equals, 0)
+	createTestPolicy(s, c)
+	c.Assert(revisionCount(s), Equals, 1)
+}
+
+func (s *configSuite) TestListPolicyRevisions(c *C) {
+	c.Assert(revisionCount(s), Equals, 0)
+
+	// edge case: the second may roll over between saving the current timestamp
+	//            and the HTTP request to upload the policy being handled
+	t := int(time.Now().Unix())
```

Comment:
If there's multiple policy uploads in the same second (whether one or many hosts), the last write wins, but there will only ever be one revision entry for it.

If you added a loop here that created multiple policies and you checked that N revisions were created, the test would fail since they'll all be created in the same second (or some in one second, some in the subsequent second, if the UNIX timestamp crossed a second boundary during the writes).